picard/disc.py

```diff
@@ -46,9 +46,13 @@ def read(self, device=None):
         if device is None:
             device = discid.get_default_device()
         log.debug("Reading CD using device: %r", device)
-        disc = discid.read(device)
-        self.id = disc.id
-        self.submission_url = disc.submission_url
+        try:
+            disc = discid.read(device)
+            self.id = disc.id
+            self.submission_url = disc.submission_url
+        except discid.disc.DiscError as e:
+            log.error("Error while reading %r: %s" % (device, str(e)))
```

Comment:
Any reason we are using %r here?

Reply:
It causes logs like `E: 16:58:17 Error while reading b'/dev/sr1': cannot open device `/dev/sr1'`

Reply:
I didn't want to bother encodings etc...